src/Module/Photo.php

```diff
@@ -83,26 +83,41 @@ public static function rawContent(array $parameters = [])
 			}
 
 			if (!empty($parameters['nickname_ext'])) {
-				$nickname = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
-				$user = User::getByNickname($nickname, ['uid']);
-				if (empty($user)) {
-					throw new HTTPException\NotFoundException();
-				}
+				if (in_array($parameters['type'], ['contact', 'header'])) {
+					$guid = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
+					$account = DBA::selectFirst('account-user-view', ['id'], ['guid' => $guid], ['order' => ['uid' => true]]);
+					if (empty($account)) {
+						throw new HTTPException\NotFoundException();
+					}
+
+					$id = $account['id'];
```

Comment:
Please create separate routes and parameters for this case instead of piling on the same parameter.

Something like:
```php
	'/photo' => [
		'/{name}'                                                     => [Module\Photo::class, [R::GET]],
		'/{type}/{id:\d+}'                                            => [Module\Photo::class, [R::GET]],

		// User Id Fallback, to remove after version 2021.12
		'/{type}/{uid_ext:\d+\..*}'                                   => [Module\Photo::class, [R::GET]],

		'/{type}/{nickname_ext}'                                      => [Module\Photo::class, [R::GET]],

		'/{type:contact|header}/{customsize:\d+}/{guid:[0-9a-f]{32}}' => [Module\Photo::class, [R::GET]],

		// Contact Id Fallback, to remove after version 2021.12
		'/{type:contact|header}/{customsize:\d+}/{contact_id:\d+}'    => [Module\Photo::class, [R::GET]],

		'/{type}/{customsize:\d+}/{id:\d+}'                           => [Module\Photo::class, [R::GET]],

		// User Id Fallback, to remove after version 2021.12
		'/{type}/{customsize:\d+}/{uid_ext:\d+\..*}'                  => [Module\Photo::class, [R::GET]],

		'/{type}/{customsize:\d+}/{nickname_ext}'                     => [Module\Photo::class, [R::GET]],
	],
```

Then the guid will be available for the relevant routes as `$parameters['guid']`. Of course, ideally we should have separate Module sub-classes for most of the different cases.

Reply:
Uuuuugh. I updated the routes.